pkg/apiserver/registry/stats/nodelatencystats/rest_test.go

```diff
@@ -29,31 +30,65 @@ import (
 )
 
 func TestREST(t *testing.T) {
-	r := NewREST()
+	fakeClock := clocktesting.NewFakeClock(time.Now())
+	r := NewREST(fakeClock)
 	assert.Equal(t, &statsv1alpha1.NodeLatencyStats{}, r.New())
 	assert.Equal(t, &statsv1alpha1.NodeLatencyStats{}, r.NewList())
 	assert.False(t, r.NamespaceScoped())
 }
 
 func TestRESTCreate(t *testing.T) {
-	summary := &statsv1alpha1.NodeLatencyStats{
-		ObjectMeta:           metav1.ObjectMeta{Name: "node1"},
-		PeerNodeLatencyStats: nil,
-	}
-	expectedObj := &statsv1alpha1.NodeLatencyStats{
-		ObjectMeta:           metav1.ObjectMeta{Name: "node1"},
-		PeerNodeLatencyStats: nil,
+	fakeClock := clocktesting.NewFakeClock(time.Now())
+	tests := []struct {
+		name        string
+		summary     *statsv1alpha1.NodeLatencyStats
+		expectedObj *statsv1alpha1.NodeLatencyStats
+	}{
+		{
+			name: "create with time added",
+			summary: &statsv1alpha1.NodeLatencyStats{
+				ObjectMeta:           metav1.ObjectMeta{Name: "node1", CreationTimestamp: metav1.Time{Time: fakeClock.Now()}},
+				PeerNodeLatencyStats: nil,
+			},
+			expectedObj: &statsv1alpha1.NodeLatencyStats{
+				ObjectMeta:           metav1.ObjectMeta{Name: "node1", CreationTimestamp: metav1.Time{Time: fakeClock.Now()}},
+				PeerNodeLatencyStats: nil,
+			},
+		},
+		{
+			name: "create with time not added",
+			summary: &statsv1alpha1.NodeLatencyStats{
+				ObjectMeta:           metav1.ObjectMeta{Name: "node1"},
+				PeerNodeLatencyStats: nil,
+			},
+			expectedObj: &statsv1alpha1.NodeLatencyStats{
+				ObjectMeta:           metav1.ObjectMeta{Name: "node1"},
+				PeerNodeLatencyStats: nil,
+			},
+		},
 	}
 
-	r := NewREST()
-	ctx := context.Background()
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			r := NewREST(fakeClock)
+			// Advance the fakeClock to simulate time advancement.
+			fakeClock.SetTime(fakeClock.Now().Add(1 * time.Minute))
+			ctx := context.Background()
+			obj, err := r.Create(ctx, tt.summary, nil, nil)
+			require.NoError(t, err)
 
-	obj, err := r.Create(ctx, summary, nil, nil)
-	require.NoError(t, err)
-	assert.Equal(t, expectedObj, obj)
+			if tt.name == "create with time not added" {
+				assert.Equal(t, metav1.Time{Time: fakeClock.Now()}, obj.(*statsv1alpha1.NodeLatencyStats).CreationTimestamp)
+
+			} else {
+				assert.Equal(t, tt.summary.CreationTimestamp, obj.(*statsv1alpha1.NodeLatencyStats).CreationTimestamp)
+			}
```

Comment:
this is not ok. You should not modify the test logic based on the test case name, otherwise it defeats the purpose of having table-driven tests. You need to use `expectedObj` which you are ignoring at the moment.